axelor-human-resource/src/main/java/com/axelor/apps/hr/service/batch/BatchTimesheetReminder.java

```diff
@@ -0,0 +1,119 @@
+import com.google.inject.Inject;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.List;
+import javax.mail.MessagingException;
+
+public class BatchTimesheetReminder extends BatchStrategy {
+
+  @Inject protected TimesheetRepository timesheetRepo;
+  @Inject protected MessageService messageService;
+
+  @Override
+  protected void process() {
+    for (Employee employee : getEmployeesWithoutRecentTimesheet()) {
+      try {
+        sendReminder(employee);
+
+      } catch (AxelorException | MessagingException | IOException e) {
+        TraceBackService.trace(new Exception(e), Employee.class.getSimpleName(), batch.getId());
+        incrementAnomaly();
+
+      } finally {
+        incrementDone();
+      }
+    }
+  }
+
+  @Override
+  protected void stop() {
+    String comment =
+        String.format(I18n.get(IExceptionMessage.BATCH_TIMESHEET_REMINDER_DONE), batch.getDone())
+            + "<br/>"
+            + String.format(
+                I18n.get(IExceptionMessage.BATCH_TIMESHEET_REMINDER_ANOMALY), batch.getAnomaly());
+
+    addComment(comment);
+    super.stop();
+  }
+
+  private List<Employee> getEmployeesWithoutRecentTimesheet() {
+    LocalDate now = LocalDate.now();
+    long daysBeforeReminder = batch.getHrBatch().getDaysBeforeReminder().longValue();
+
+    List<Employee> employees =
+        employeeRepository
+            .all()
+            .filter(
+                "self.timesheetReminder = 't' AND self.mainEmploymentContract.payCompany = :companyId")
+            .bind("companyId", batch.getHrBatch().getCompany().getId())
+            .fetch();
+    employees.removeIf(employee -> hasRecentTimesheet(now, daysBeforeReminder, employee));
+    return employees;
+  }
+
+  private boolean hasRecentTimesheet(LocalDate now, long daysBeforeReminder, Employee employee) {
+    Timesheet timesheet =
+        timesheetRepo
+            .all()
+            .filter(
+                "self.user.id = :userId AND self.statusSelect IN (:confirmed, :validated) AND self.company = :companyId")
+            .bind("userId", employee.getUser().getId())
+            .bind("confirmed", TimesheetRepository.STATUS_CONFIRMED)
+            .bind("validated", TimesheetRepository.STATUS_VALIDATED)
+            .bind("companyId", batch.getHrBatch().getCompany().getId())
+            .order("-toDate")
+            .fetchOne();
+    return timesheet != null && timesheet.getToDate().plusDays(daysBeforeReminder).isAfter(now);
+  }
+
+  private void sendReminder(Employee employee)
+      throws AxelorException, MessagingException, IOException {
+    Message message = new Message();
+    message.setMediaTypeSelect(MessageRepository.MEDIA_TYPE_EMAIL);
+    message.setReplyToEmailAddressSet(new HashSet<>());
+    message.setCcEmailAddressSet(new HashSet<>());
+    message.setBccEmailAddressSet(new HashSet<>());
+    message.addToEmailAddressSetItem(employee.getContactPartner().getEmailAddress());
```

Comment:
@ale-axelor 
`employee` cannot be null because the method is called from a for loop interating other employees. And `contactPartner` has `@NotNull` annotation.